src/counter.rs

```diff
@@ -134,10 +134,11 @@ mod tests {
         counter.inc_by(42.0).unwrap();
         assert_eq!(counter.get() as u64, 43);
 
-        let mf = counter.collect();
-        let m = mf.get_metric().as_ref().get(0).unwrap();
-        assert_eq!(m.get_label().len(), 2);
-        assert_eq!(m.get_counter().get_value() as u64, 43);
+        for mf in counter.collect() {
+            let m = mf.get_metric().as_ref().get(0).unwrap();
+            assert_eq!(m.get_label().len(), 2);
+            assert_eq!(m.get_counter().get_value() as u64, 43);
+        }
```

Comment:
Since counter only has 1 metric, I think it's better to test it like this:
```
let mf = counter.collect();
assert_eq!(mf.len(), 1);
let m = mf[0].get_metric().as_ref().get(0).unwrap();
...
```